node/router/src/lib.rs

```diff
@@ -227,6 +227,12 @@ impl<N: Network> Router<N> {
                 }
                 disconnected
             } else {
+                // FIXME (ljedrz): this shouldn't be necessary; it's a double-check
+                //  that the higher-level collection is consistent with the resolver.
+                if router.is_connected(&peer_ip) {
+                    warn!("Fallback connection artifact cleanup (report this to @ljedrz)");
+                    router.remove_connected_peer(peer_ip);
+                }
```

Comment:
@ljedrz this approach seems like a hack. Can you confirm you've seen empirically ghost IPs/peers left over in the router?

Reply:
yes, I've seen them in the Canary network, hence this adjustment; last time I checked the logs there were only a handful or less, but it warrants a closer inspection, and justifies this fallback

Reply:
We should open an issue for that, so we don't forget to fix it and to eventually remove those lines.

Reply:
https://github.com/ProvableHQ/snarkOS/issues/3705